Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm

```diff
@@ -527,18 +527,20 @@ static bool sampleBufferRenderersSupportKeySession()
 
 void SourceBufferPrivateAVFObjC::didUpdateFormatDescriptionForTrackId(Ref<TrackInfo>&& formatDescription, uint64_t trackId)
 {
-    if (is<VideoInfo>(formatDescription)) {
-        auto result = m_videoTracks.find(AtomString::number(trackId));
-        if (result != m_videoTracks.end())
-            result->value->setFormatDescription(downcast<VideoInfo>(formatDescription.get()));
-        return;
-    }
+    SourceBufferPrivate::didUpdateFormatDescriptionForTrackId(WTFMove(formatDescription), trackId, [weakThis = WeakPtr { *this }, this](Ref<TrackInfo>&& formatDescription, uint64_t trackId) {
+        if (is<VideoInfo>(formatDescription)) {
```

Comment:
protectedThis is probably better but if we need weakThis, let's add a `if (weakThis)` check.

Reply:
done, this is called from code also checking WeakPtr and so this can't be called, but I added the test. All this code will be gone soon too